File: app/src/main/java/com/olokogini/moriai/ui/main/chat/data/ChatDao.java
package com.olokogini.moriai.ui.main.chat.data;

import androidx.room.*;
import java.util.List;
import kotlinx.coroutines.flow.Flow;


@Dao
public interface ChatDao {

    @Query("SELECT * FROM chat_message ORDER BY timestamp ASC")
    kotlinx.coroutines.flow.Flow<java.util.List<ChatMessageEntity>> getMessages();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertMessage(ChatMessageEntity message);

    @Query("DELETE FROM chat_message")
    void clearChat();
}

File: app/src/main/java/com/olokogini/moriai/ui/main/chat/data/ChatDao.kt
package com.olokogini.moriai.ui.main.chat.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface ChatDao {

    @Query("SELECT * FROM chat_message ORDER BY timestamp ASC")
    fun getMessages(): Flow<List<ChatMessageEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertMessage(message: ChatMessageEntity)

    @Query("DELETE FROM chat_message")
    suspend fun clearChat()
}